aspnetcore/mvc/razor-pages/razor-pages-authorization-conventions/sample/Pages/Index.cshtml

```diff
@@ -0,0 +1,121 @@
+﻿@page
+@model IndexModel
+@inject Microsoft.AspNetCore.Http.IHttpContextAccessor HttpContextAccessor;
+@{
+    ViewData["Title"] = "Home page";
+}
+
+@if (HttpContextAccessor.HttpContext.User.Identity.IsAuthenticated)
```

Comment:
There's an `User` property on the page that you should be able to use instead. No need to DI in the accessor.